services/MerchStoreService.ts

```diff
@@ -154,19 +177,24 @@ export default class MerchStoreService {
       const item = await merchItemRepository.findByUuid(uuid);
       if (!item) throw new NotFoundError();
       const hasBeenOrdered = await Repositories.merchOrderItem(txn).hasItemBeenOrdered(uuid);
-      if (hasBeenOrdered) throw new UserError('This item has been ordered already');
+      if (hasBeenOrdered) throw new UserError('This item has been ordered and cannot be deleted');
       return merchItemRepository.deleteMerchItem(item);
     });
   }
 
   public async createItemOption(item: Uuid, option: MerchItemOption): Promise<PublicMerchItemOption> {
     return this.transactions.readWrite(async (txn) => {
       const merchItem = await Repositories.merchStoreItem(txn).findByUuid(item);
-      if (!merchItem) throw new NotFoundError('Item not found');
+      if (!merchItem) throw new NotFoundError('Merch item not found');
+      if (!merchItem.hasVariantsEnabled) throw new UserError('Cannot add option to items with variants disabled');
+      const hasDifferentOptionType = merchItem.options[0].metadata?.type !== option.metadata?.type;
+      if (hasDifferentOptionType) throw new UserError('Item cannot have multiple option types');
+
       const merchItemOptionRepository = Repositories.merchStoreItemOption(txn);
       const createdOption = MerchandiseItemOptionModel.create({ ...option, item: merchItem });
       await merchItemOptionRepository.upsertMerchItemOption(createdOption);
-      return merchItemOptionRepository.findByUuid(createdOption.uuid);
+      const upsertedOption = await merchItemOptionRepository.findByUuid(createdOption.uuid);
```

Comment:
Doesn't `upsertMerchItemOption` return `upsertedOption`?

Reply:
Yes but we generally return the "public" version of the objects i.e. `PublicMerchItemOption`

Reply:
Wait never mind either way we don't need that call whoops, my b